chrome/background.js

```diff
@@ -99,96 +100,86 @@ function onMessage(request, sender, sendResponse) {
   // spawn a new tab with pre-provided credentials
   if (request.action == "launch") {
     chrome.tabs.create({ url: request.url }, function(tab) {
-      var tabLoaded = false;
       var authAttempted = false;
 
-      // listener function for authentication interception
-      function authListener(requestDetails) {
+      authListeners[tab.id] = function(requestDetails) {
         // only supply credentials if this is the first time for this tab, and the tab is not loaded
         if (authAttempted) {
           return {};
         }
         authAttempted = true;
-
-        // don't supply credentials for loaded tabs
-        if (tabLoaded) {
-          return {};
-        }
-
-        // ask the user before sending credentials to a different domain
-        var launchHost = request.url.match(/:\/\/([^\/]+)/)[1];
-        if (launchHost !== requestDetails.challenger.host) {
-          var message =
-            "You are about to send login credentials to a domain that is different than " +
-            "the one you lauched from the browserpass extension. Do you wish to proceed?\n\n" +
-            "Launched URL: " +
-            request.url +
-            "\n" +
-            "Authentication URL: " +
-            requestDetails.url;
-          if (!confirm(message)) {
-            return {};
-          }
-        }
-
-        // ask the user before sending credentials over an insecure connection
-        if (!requestDetails.url.match(/^https:/i)) {
-          var message =
-            "You are about to send login credentials via an insecure connection!\n\n" +
-            "Are you sure you want to do this? If there is an attacker watching your " +
-            "network traffic, they may be able to see your username and password.\n\n" +
-            "URL: " +
-            requestDetails.url;
-          if (!confirm(message)) {
-            return {};
-          }
-        }
-
-        // supply credentials
-        return {
-          authCredentials: {
-            username: request.username,
-            password: request.password
-          }
-        };
-      }
+        onAuthRequired(request, requestDetails);
+      };
 
       // intercept requests for authentication
       chrome.webRequest.onAuthRequired.addListener(
-        authListener,
+        authListeners[tab.id],
         { urls: ["*://*/*"], tabId: tab.id },
         ["blocking"]
       );
-
-      // notice when the tab has been loaded
-      chrome.tabs.onUpdated.addListener(function tabCompleteListener(
-        tabId,
-        info
-      ) {
-        if (tabId == tab.id && info.status == "complete") {
-          // remove listeners
-          chrome.tabs.onUpdated.removeListener(tabCompleteListener);
-          chrome.webRequest.onAuthRequired.removeListener(authListener);
-
-          // mark tab as loaded & wipe credentials
-          tabLoaded = true;
-          request.username = request.password = null;
```

Comment:
The reference to this object is in this function and now in `onAuthRequired`, as soon as both functions are gone the credentials are not accessible anyway - is there a point in additionally wiping them here?

Reply:
As above - this was leftover from an earlier iteration of the bugfix; it can go.